services/vaults/holdings/save/vaults.js

```diff
@@ -0,0 +1,293 @@
+'use strict';
+
+const config = require('./config.js');
+
+
+module.exports = [
+  {
```

Comment:
Can we get vaults from vaults endpoint instead? The vaults endpoint pulls from the vault registry

Reply:
ah yes, I saw you improved that part, I think I can do this easily.

Reply:
Icannot use vaults from DB as the ABI is not stored. I could use:
`
const fetchAbi = async (address) => {
  const url = `https://api.etherscan.io/api?module=contract&action=getabi&address=${address}&apikey=${etherscanApiKey}`;
  const resp = await fetch(url).then((res) => res.json());
  const abi = JSON.parse(resp.result);
  await delay(delayTime);
  return abi;
};
`
from services/vaults/holdings/save/handler.js, but we would rely on etherscan API.

Reply:
As discussed in discord this cannot be done for now. We cannot fetch the ABI on-chain. Also, the services/vaults/holdings/save/vaults.js file is defining where the ABI is coming from. Also I cannot remove the vault.js file as, for instance the aLink and Link strategies have 6 decimals and there is no on-chain "decimals" method for these strategies for me to fetch the correct decimals. So for now we need to rely on the code in the files and not on-chain data :S